src/errors/localImportErrors.ts

```diff
@@ -10,14 +10,14 @@ export class InvalidFilePathError implements SourceError {
   constructor(public filePath: string) {}
 
   public explain() {
-    const validNonAlphanumericChars = Object.keys(charEncoding)
+    const validNonAlphanumericChars = Object.keys(nonAlphanumericCharEncoding)
       .map(char => `'${char}'`)
       .join(', ')
-    return `'${this.filePath}' must only contain alphanumeric chars or one of ${validNonAlphanumericChars}.`
+    return `'${this.filePath}' must only contain alphanumeric chars or one of ${validNonAlphanumericChars}, and must not contain consecutive slashes '//'.`
```

Comment:
```suggestion
    return `File path '${this.filePath}' must only contain alphanumeric chars or one of ${validNonAlphanumericChars}, and must not contain consecutive slashes ('//').`
```
Minor rewording.
As an aside, we can consider splitting this into two separate errors, one for file path containing invalid characters and one for consecutive slashes, if it helps reduce the length of error messages/makes the error clearer to students.

Reply:
Wouldn't adding `File path` at the start make the length of the error message even longer? I'm not sure what value it has since it's rather obvious that the file path is a file path (nothing else would be named with slashes in them).

Regarding the addition of brackets, this means that the the error message will be something like `must only contain one of '_', '/', '.', '-', and must not contain consecutive slashes ('//')`. Is this inconsistency intended?

Splitting the error message would require having two different kinds of errors that deal with invalid file paths and moving the error handling logic into the file path validation function which isn't as clean. I agree that it makes for a better user experience though. Let me think about this more.

Reply:
Adding `File path` to the error message makes it clearer that the rules apply to all file paths, so that students are less likely to make the same errors for other file paths. This also makes the error message read more similarly to other error messages in Source. (e.g. `Redeclaring name ${this.name}.` in `VariableRedeclaration`).

The inconsistency is intended as the former is referring to the characters directly but the latter describes the characters first, hence the brackets were added to indicate that the term "consecutive slashes" and the charactes `//` refer to the same thing. Perhaps changing it to `(e.g. '//')` would make it clearer?

Speaking of, `one of '_', '/', '.', '-',` can also be reworded to `the chars '_', '/', '.' or '-',`. Though this means there will be two "or"s in the sentence, it reduces the chances of misinterpreting the error message that the file path can only contain one of the special chars.

Reply:
> Adding `File path` to the error message makes it clearer that the rules apply to all file paths, so that students are less likely to make the same errors for other file paths. This also makes the error message read more similarly to other error messages in Source (e.g. `Redeclaring name ${this.name}.` in `VariableRedeclaration`).

Ok, that's fair.

> The inconsistency is intended as the former is referring to the characters directly but the latter describes the characters first, hence the brackets were added to indicate that the term "consecutive slashes" and the characters `//` refer to the same thing. Perhaps changing it to `(e.g. '//')` would make it clearer?

My understanding is that "e.g." is semantically the same as "for example". Are you sure about this?

> Speaking of, `one of '_', '/', '.', '-',` can also be reworded to `the chars '_', '/', '.' or '-',`. Though this means there will be two "or"s in the sentence, it reduces the chances of misinterpreting the error message that the file path can only contain one of the special chars.

This is slightly problematic as it requires special formatting instead of just calling `join` on the valid non-alphanumeric characters. But I guess I can implement a formatting function if it will make the user experience better.

Reply:
Yeah on second thoughts let's not add the "e.g.". The rewording of `one of '_', '/', '.', '-',` is also not really needed. The initial concern was the inconsistency between the wording with respect to `'_', '/', '.', '-',` and `//`, but if the errors are to be split into two then this becomes less of a concern.